token-client/src/main/java/com/sap/cloud/security/xsuaa/mtls/SpringHttpClient.java

```diff
@@ -0,0 +1,46 @@
+package com.sap.cloud.security.xsuaa.mtls;
+
+import com.sap.xsa.security.container.ClientIdentity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * SpringHttpClient provides factory methods to initialize RestTemplate for
+ * certificate(HTTPS) based and client secret(HTTP) based communications.
+ */
+public class SpringHttpClient {
+
+	private SpringHttpClient() {
+	}
+
+	/**
+	 * Creates a HTTP RestTemplate
+	 * 
+	 * @return RestTemplate instance
+	 */
+	public static RestTemplate create() {
+		return new RestTemplate();
+	}
+
+	/**
+	 * Creates a HTTPS RestTemplate. Used to setup HTTPS client for X.509
+	 * certificate based communication. Derives certificate and private key values
+	 * from ClientIdentity.
+	 *
+	 * @param clientIdentity
+	 *            ClientIdentity of Xsuaa Service
+	 * @return RestTemplate instance
+	 * @throws ServiceClientException
+	 *             if ClientIdentity wasn't certificate based
+	 */
+	public static RestTemplate create(ClientIdentity clientIdentity) throws ServiceClientException {
+		if (clientIdentity.isCertificateBased()) {
+			HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
+			requestFactory.setHttpClient(HttpClient.create(clientIdentity));
+
+			return new RestTemplate(requestFactory);
+		}
+
+		throw new ServiceClientException("ClientIdentity is not certificate based, can't initiate mTLS http client");
```

Comment:
HttpClient and SpringHttpClient does not behave similar. When remembering right HttpClient returns default client in case clientIdentity is not cert based and only raises exception in case of IOException...

Reply:
yap, good finding, adjusted to match HttpClient and SpringHttpClient